oauth2/oauth2-core/src/main/java/org/springframework/security/oauth2/core/user/OAuth2UserAuthority.java

```diff
@@ -44,6 +45,8 @@ public class OAuth2UserAuthority implements GrantedAuthority {
 
 	private final String userNameAttributeName;
```

Comment:
This property should be deprecated as should all the methods/constructors that use it.

Please also address `OidcUserAuthority` which extends from this class. It will likely need it's own `Builder` that extends from this `Builder` and have an overridden `withUsername` static method.